### DISCUSSION TITLE
Cyberduck and Mountain Duck reconnection issues when network reconnects

### DISCUSSION THREAD

**Discussion by simbaoyee** [iterate-ch/cyberduck#15230] [Q&A]
I have Mountain Duck and Cyberduck bookmarks set to auto connect when windows starts.
Now, when windows starts and network connection is present, the apps start up and connect to cloud storages correctly.
However, when there is no internet connection present at time of windows  booting up, Mountain Duck and Cyber Duck go berserk.
They display a popup asking to try again.
When internet is restored and I try to connect again, instead of using saved bookmarks,, the apps fire up a new browser window and ask me to login again.
What apain.

**Comment by tatze96** [comment]
Can you please [enable debug logging](https://docs.cyberduck.io/cyberduck/support/#debug-logging-level), recreate the issue, and send us the log file afterward? Also, please add a screenshot of the login error that prompts you to re-authenticate after the manual re-connect.

If you don't like to see sensitive data to be shared publicly, please send the log file with the ticket reference GH-15230 to support@cyberduck.io, otherwise attach it here.

**Sidenote**: If you want to share the log file and the screenshots here, make sure to attach them through the GitHub web interface, as they wouldn't be visible to us otherwise.

**Comment by simbaoyee** [comment]
Here is the log file and a video of the login screen.
[mountainduck.log](https://github.com/iterate-ch/cyberduck/files/13188607/mountainduck.log)

https://github.com/iterate-ch/cyberduck/assets/148552837/9f76041f-0ef0-4744-a856-d9910d082a48

One more issue. Even after the manual login is successful, the login screen stays the same giving the impression that login is still in process.

**Comment by dkocher** [comment]
The error message from `login.microsoftonline` is 
```
AADSTS70000: The provided value for the input parameter 'refresh_token' or 'assertion' is not valid. The token was issued for a different client id
```

We use the same OAuth Client ID for both _Cyberduck_ and _Mountain Duck_ that should allow sharing tokens.

**Reply by dkocher** [reply]
I have opened #15284 to track this problem.

**Comment by simbaoyee** [comment]
Thanks.

**Comment by tatze96** [comment]
You can try to resolve the authentication issue by resetting the saved OAuth tokens. To do so, delete all `duck:onedrive?user=(user)` entries out of the *Windows Credentials* section of the *Windows Credential Manager*.

Ensure that Cyberduck and Mountain Duck aren't running while removing the entries.

**Reply by Wrong-Code** [reply]
Done that, it doesn't last longer than some hours. The day after the tokens for OneDrive, Google Drive and Box services are again not valid, Really annoying,

**Reply by dkocher** [reply]
Opened #15309 to track this issue.

**Comment by simbaoyee** [comment]
will try and update
